### PR TITLE
feat: #210 add swagger

### PR DESCRIPTION
**Pull request by Wongcheukwai**
Changes are:
1. add swaggerOption.js in utils
2. add it in app.js
3. give an example of signup in authRoutes.js
To use:
a. install using: npm install swagger-ui-express swagger-jsdoc
b. if you want to add swagger comment, please refer to the example in authRoutes.js (Please be noted the example is generated by generative AI models).
c. open http://localhost:3000/api-docs to check Swagger API.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ connectDB();
 // Setup middleware
 setupMiddleware(app);
 
+// Use Swagger
+swaggerInstall(app);
+
```
Comment (ucfyao) [inline app.js:23]: why install

File: app/utils/swagger.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const swaggerJsdoc = require('swagger-jsdoc');
+const swaggerUI = require('swagger-ui-express');
+const config = require('../../config');
+
+const swaggerOptions = {
+  swaggerDefinition: {
+    openapi: '3.0.0',
+    info: {
+      title: 'API of Xiaobao',
+      version: '1.0.0',
```
Comment (ucfyao) [inline app/utils/swagger.js:10]: api of moow

File: app/utils/swagger.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      description: 'API documentation for Xiaobao.io',
+    },
+    servers: [
```
Comment (ucfyao) [inline app/utils/swagger.js:13]: moow.cc

File: app/utils/swagger.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  
+const swaggerSpecs = swaggerJsdoc(swaggerOptions);
+
+const swaggerInstall = function (app) {
+    if (!app) {
+      app = express();
+    }
+    app.get("/swagger.json", swaggerJson);
+    if (config.env === 'test') {
+    app.use("/api-docs", swaggerUI.serve, swaggerUI.setup(swaggerSpecs));
```
Comment (ucfyao) [inline app/utils/swagger.js:38]: dev?

File: app.js (hunk 1 above)
Comment (Wongcheukwai) [reply app.js:23]: change it to initial

**Review by ucfyao — APPROVED** [review]

**Pull request merged**